decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md

```diff
@@ -0,0 +1,361 @@
+contract of its parent. For example, that all alembic files are a geometry cache, and that all
+geometry caches might have coordinate system based data, and that all geometry caches are a type of
+binary blob. This could be described in a specification: entity. blob.geometry._cache.alembic.
+Another example could be an exr render, which could be a type of image, which could also be a type
+of blob binary data, represented as: entity.blob.image.exr .
+
+#### Pros
+
+- only one way to represent the specification
+- automatic fall back mechanism through inheritance
+- proven in real-world, working scenarios
+
+#### Cons
+
+- difficult to insert new specifications in between types, requires updating the child type
+  identifier of the new insert
+- requires knowledge of as many possible entities as possible to be efficiently taxonomied into an
+  efficient and useful hierarchy
+- sharing behaviours between taxonomically unrelated types - for example a frame range applied to
+  both a geometry cache and an image - is difficult, it relies on duplicated property definitions
+  on each of specifications, meaning that the trait of “having a frame range becomes” implied
+  though well-known property names, rather than an explicitly stated trait.
+- in order to share well-known base specification hierarchies with the OpenAssetIO
+  community,interop is facilitated through the use of common base specifications. If these base
+  hierarchies change, every copy of the specifications will no longer be able to interact with each
+  other despite possibly having the same expected behaviour
+
+### Option 2
+
+Trait based compositional specifications. Specifications are made from generic traits that
+guarantee certain behaviours or consumption patterns from the Manager or Host. The alembic cache
+could be represented as the following: entity.geometry.cache. The exrs rendered out from the Option
+1 example could also be represented as entity.image.cache as they are a type of image cache.
+Managers that are interested in freeing up some memory could then query entity.cache and schedule
+them for deletion.
+
+#### Pros
+
+- adding to specifications is easy by adding traits
+- traits can be shared between specifications easily without needing to hide them as emergent
+  phenomena based on well-known property names
+- common traits can be shared with the OpenAssetIO community without having to exactly match a
+  whole specification, decreasing the risk of inclusion of very well known traits (such as `blob`)
+  in the core API if required.
+
+#### Cons
+
+- no guarantee that the traits aren't contradictory
+- working with traits could be more complex than working with a single parent hierarchy due to
+  multiple combinations of traits being possible
+- duplication of traited specification is possible
+- not yet practically validated
+- changes to specific traits still affects all specifications that incorporate them
+
+Estimated cost: Small|Medium|Large
```

Comment:
[fix] Needs cost selecting

Reply:
Added cost estimation: e7717b77781fa9927a5dd0e23863ca706d113557

Reply:
Ace thanks! I think the template needs up dating, as if we have on option 2 - we'd need one on option 1 for comparison. If effort isn't a consideration, it can be removed from both.

Reply:
Removed the effort ad33b2a